node/service/src/parachains_db/upgrade.rs

```diff
@@ -102,3 +125,79 @@ fn migrate_from_version_0_to_1(path: &Path) -> Result<(), Error> {
 
 	Ok(())
 }
+
+// This currently clears columns which had their configs altered between versions.
+// The columns to be changed are constrained by the `allowed_columns` vector.
+fn paritydb_fix_columns(
+	path: &Path,
+	options: parity_db::Options,
+	allowed_columns: Vec<u32>,
+) -> io::Result<()> {
+	// Figure out which columns to delete. This will be determined by inspecting
+	// the metadata file.
+	if let Ok(Some(metadata)) = parity_db::Options::load_metadata(&path) {
```

Comment:
should we propagate the error if we loading metadata errors?

Reply:
I think you are right, now this would only be called if the db exists.